site/public/js/forum.js

```diff
@@ -754,10 +754,11 @@ function showEditPostForm(post_id, thread_id, shouldEditThread, render_markdown,
             var anon = json.anon;
             var change_anon = json.change_anon;
             var user_id = escapeSpecialChars(json.user);
+            json.post_time = json.post_time.replace(" ", "T");
```

Comment:
Where is this `post_time` coming from and can we update whatever format it is there to make `Date.parse` happy?

Reply:
I'm not sure where it's coming from.  I tried to track it down but there isn't much to work with.  I'm not sure which file is the backend for this.  The only issue in the date is that there is a space between the date and the timestamp.  JS expects a T in front of the timestamp so I simply replaced the space with a T.  It could be a difference in how PHP and JS process dates.  The general format is exactly the same except for that.  PHP also adds a higher precision for the seconds than JS but JS seems to handle it just fine.

Reply:
To help track this down, we can see that this is firing on making a request to `buildCourseUrl(['forum', 'posts', 'get']);` where that gives us a url path of `{_semester}/{_course}/forum/posts/get` within the code. Searching the codebase for that route leads us to the function [`getEditPostContent`](https://github.com/Submitty/Submitty/blob/dbee91d58f4918b5c9fbc60bae2e5989ac959f73/site/app/controllers/forum/ForumController.php#L1079-L1104). Within that, we see then that the returned JSON object is coming from the [`getPost`](https://github.com/Submitty/Submitty/blob/dbee91d58f4918b5c9fbc60bae2e5989ac959f73/site/app/libraries/database/DatabaseQueries.php#L724) database method, where timestamp is just the string as it comes out of the database. Scanning the database, the DB query is used in a handful of places, while this route is only used here, so we can safely do that replace within the controller method. As part of this, we can also properly amend the the `00` to it there as well. The easiest way would be to parse `$result['timestamp'];` into a `\DateTime` and then format it to something that can be directly used within the JS.

We want to avoid doing these sort of manipulations in JS as it's a weak link between PHP / JS and so prone to possible break down the road when a developer inadvertently changes the upstream function, as well as that we're trying to work around vendor quirks here which is a recipe for disaster.